src/main/java/com/amazon/dlic/auth/http/jwt/HTTPExtensionJwtAuthenticator.java

```diff
@@ -0,0 +1,272 @@
+            }
+        }
+
+        try {
+            final Claims claims = jwtParser.parseClaimsJws(jwtToken).getBody();
+
+            final String subject = extractSubject(claims, request);
+
+            final String audience = claims.getAudience();
+
+            //TODO: GET ROLESCLAIM DEPENDING ON THE STATUS OF BWC MODE. ON: er / OFF: dr
+            Object rolesObject = null;
+            String[] roles = null;
+
+            if (bwcPluginCompatibilityMode) {
+                rolesObject = claims.get("dr");
+            } else if (!bwcPluginCompatibilityMode) {
+                rolesObject = claims.get("er");
+            }
+
+            if (rolesObject == null) {
+                log.warn(
+                        "Failed to get roles from JWT claims. Check if this key is correct and available in the JWT payload.");
+                roles = new String[0];
+            } else {
+                final String rolesClaim = rolesObject.toString();
+                //IF THE USER IS IN BWC MODE WE ARE EXPECTING UNENCRYPTED ROLES. IF THE BWC MODE IS OFF WE ARE EXPECTING ENCRYPTED ROLES
+                if (!bwcPluginCompatibilityMode) {
+                    //TODO: WHERE TO GET THE ENCRYTION KEY
+                    final String decryptedRoles = EncryptionDecryptionUtil.decrypt(encryptionKey, rolesClaim);
+                    roles = Arrays.stream(decryptedRoles.split(",")).map(String::trim).toArray(String[]::new);
+                } else if (bwcPluginCompatibilityMode){
+                    roles = Arrays.stream(rolesClaim.split(",")).map(String::trim).toArray(String[]::new);
+                }
+            }
+
+            if (subject == null) {
+                log.error("No subject found in JWT token");
+                return null;
+            }
+
+            if (audience == null) {
+                log.error("No audience found in JWT token");
+            }
+
+            final AuthCredentials ac = new AuthCredentials(subject, roles).markComplete();
+
+            for(Entry<String, Object> claim: claims.entrySet()) {
+                ac.addAttribute("attr.jwt."+claim.getKey(), String.valueOf(claim.getValue()));
+            }
+
+            return ac;
+
+        } catch (WeakKeyException e) {
+            log.error("Cannot authenticate user with JWT because of ", e);
+            return null;
+        } catch (Exception e) {
+            if(log.isDebugEnabled()) {
+                log.debug("Invalid or expired JWT token.", e);
+            }
+            return null;
+        }
+    }
+
+    @Override
+    public boolean reRequestAuthentication(final RestChannel channel, AuthCredentials creds) {
+        final BytesRestResponse wwwAuthenticateResponse = new BytesRestResponse(RestStatus.UNAUTHORIZED,"");
+        wwwAuthenticateResponse.addHeader("WWW-Authenticate", "Bearer realm=\"OpenSearch Security\"");
+        channel.sendResponse(wwwAuthenticateResponse);
+        return true;
```

Comment:
Instead lets return false like in the different proxy authenticators

Reply:
Fixed!